course/frontend/array-shuffle/scripts/array-shuffle-polyfill.js

```diff
@@ -1,32 +1,22 @@
-/**
- * Array.shuffle() Crea un clon exacto de un array y desordena aleatoriamente sus elementos.
- *
- * @code
- *  var originalArray = [1,2,3,'a','b','c'];
- * 	var newArray = originalArray.shuffle();
- * @endcode
- *
- * @return {array} newArray. Retorna un nuevo array.
- */
-Array.prototype.shuffle = function () {
-	// for (var i = 0; i < this.length; i++) {
-	// 	newArray.push(this[i]);
-	// }
-	var newArray = this.slice();
-	// newArray == just a clone of "this" (the original array)!
-	var currentIndex = newArray.length;
-	var tempValue, randomIndex;
-	while (0 !== currentIndex) {
-		randomIndex = Math.floor(Math.random() * currentIndex); // cojo un elemento al azar
-		currentIndex -= 1;
-		tempValue = newArray[currentIndex]; // Me guardo el valor de la posición "A"
-		newArray[currentIndex] = newArray[randomIndex]; // Copio el valor de "B" en la posición "A"
-		newArray[randomIndex] = tempValue; // Copio el valor de "A" en la posición "B"
-	}
-	return newArray;
-}
-
-// WARN! this is required to avoid 'shuffle' to be present in an array for-in loop!!!
-Object.defineProperty(Array.prototype, 'shuffle', {
-	enumerable: false
-});
\ No newline at end of file
+
+Array.prototype.shuffle = function() {
```

Comment:
WELL done! but please, do not delete the solution provided before. KEEP both solutions, and comment with a low-dash the one you want to "comment" (e.g. Array.prototype._shuffle).

Reply:
You all right! It's better better modify than destroy. I include the lost lines.